packages/cli/src/index.ts

```diff
@@ -12,183 +12,285 @@ import {
-  .alias('dev')
-  .option('-l, --lazy', 'lazyCompilation')
-  .option('--host <host>', 'specify host')
-  .option('--port <port>', 'specify port')
-  .option('--open', 'open browser on server start')
-  .option('--hmr', 'enable hot module replacement')
-  .option('--cors', 'enable cors')
-  .option('--strictPort', 'specified port is already in use, exit with error')
-  .action(
-    async (
-      rootPath: string,
-      options: FarmCLIServerOptions & GlobalFarmCLIOptions
-    ) => {
-      const { root, configPath } = resolveCliConfig(rootPath, options);
-      const resolveOptions = resolveCommandOptions(options);
-
-      const defaultOptions = {
-        root,
-        compilation: {
-          lazyCompilation: options.lazy
-        },
-        server: resolveOptions,
-        clearScreen: options.clearScreen,
-        configPath,
-        mode: options.mode
-      };
-
-      const { start } = await resolveCore();
-      handleAsyncOperationErrors(
-        start(defaultOptions),
-        'Failed to start server'
-      );
+const devCommand = defineCommand({
+  meta: {
+    name: 'dev',
+    description:
+      'Compile the project in dev mode and serve it with farm dev server'
+  },
+  args: {
+    root: {
+      type: 'positional',
+      description: 'root path',
+      required: false,
+      valueHint: 'path'
+    },
+    lazy: { type: 'boolean', alias: 'l', description: 'lazyCompilation' },
+    host: { type: 'string', description: 'specify host' },
+    port: { type: 'string', description: 'specify port' },
+    open: { type: 'boolean', description: 'open browser on server start' },
+    hmr: { type: 'boolean', description: 'enable hot module replacement' },
+    cors: { type: 'boolean', description: 'enable cors' },
+    strictPort: {
+      type: 'boolean',
+      description: 'specified port is already in use, exit with error'
     }
-  );
-
-// build command
-cli
-  .command('build [root]', 'compile the project in production mode')
-  .option('-o, --outDir <dir>', 'output directory')
-  .option('-i, --input <file>', 'input file path')
-  .option('-w, --watch', 'watch file change')
-  .option('--target <target>', 'transpile targetEnv node, browser')
-  .option('--format <format>', 'transpile format esm, commonjs')
-  .option('--sourcemap', 'output source maps for build')
-  .option('--treeShaking', 'Eliminate useless code without side effects')
-  .option('--minify', 'code compression at build time')
-  .action(
-    async (
-      rootPath: string,
-      options: FarmCLIBuildOptions & GlobalFarmCLIOptions
-    ) => {
-      const { root, configPath } = resolveCliConfig(rootPath, options);
-
-      const defaultOptions = {
-        root,
-        configPath,
-        ...getOptionFromBuildOption(options)
-      };
-
-      const { build } = await resolveCore();
-      handleAsyncOperationErrors(build(defaultOptions), 'error during build');
+  },
+  async run({ args }: { args: FarmCLICommonOptions & FarmCLIServerOptions }) {
```

Comment:
The alias parameter in arg is not configured. Is there a way to configure shared arg in citty? 
If I use -m now it doesn't work